execution/gethexec/express_lane_service_test.go

```diff
@@ -390,26 +400,47 @@ func Test_expressLaneService_sequenceExpressLaneSubmission_outOfOrder(t *testing
 			Transaction:    emptyTx,
 		},
 	}
+	var wg sync.WaitGroup
+	wg.Add(7)
 	for _, msg := range messages {
-		err := els.sequenceExpressLaneSubmission(ctx, msg)
-		require.NoError(t, err)
+		go func(w *sync.WaitGroup) {
+			w.Done()
+			err := els.sequenceExpressLaneSubmission(ctx, msg)
+			if msg.SequenceNumber != 10 { // Because this go-routine will be interrupted after the test itself ends and 10 will still be waiting for result
+				require.NoError(t, err)
+				w.Done()
+			}
+		}(&wg)
 	}
+	wg.Wait()
+
 	// We should have only published 2, as we are missing sequence number 3.
+	time.Sleep(2 * time.Second)
```

Comment:
Is this sleep needed since `wg.Wait` should wait until 1 and 2 are published?

Reply:
The sleep is needed because wg.Done() will guarantee the launch of goroutine but not the submission of transactions themself. 

Here's a scenario where we would need it- goroutines with msg 1 and 2 gets launched and accepted so the wg count now becomes 7-4 = 3, and the we launch goroutines for rest of the messages which are future seq numbers thus they will get queued, but as w.Done() is called before els.sequenceExpressLaneSubmission(ctx, msg) so wg.Wait() would return but the future seq num messages might not be yet added, which is not what we want here.  When we check for 
```
els.Lock()
require.Equal(t, 3, len(els.msgAndResultBySequenceNumber)) // Processed txs are deleted
els.Unlock()
```

Reply:
No worries, it's not always so feasible to remove the sleeps.